lib/src/source/git.dart

```diff
@@ -306,10 +306,13 @@ class BoundGitSource extends CachedSource {
       var revisionCachePath = _revisionCachePath(id);
       await _revisionCacheClones.putIfAbsent(revisionCachePath, () async {
         if (!entryExists(revisionCachePath)) {
-          await _clone(_repoCachePath(ref), revisionCachePath);
+          await _clone(ref.description['url'], revisionCachePath,
+              reference: _repoCachePath(ref));
           await _checkOut(revisionCachePath, id.description['resolved-ref']);
+          await _pullLFS(revisionCachePath);
           _writePackageList(revisionCachePath, [id.description['path']]);
         } else {
+          await _pullLFS(revisionCachePath);
```

Comment:
Why do we need this? If `entryExists(revisionCachePath)` then can't we assume `_pullLFS` has already completed?

I agree that for existing clones this won't be the case. So if someone cloned a repository using LFS before LFS support was added to `dart pub get` then it won't work.

Solution for this problem could be to simply run `dart pub cache repair`.